### PR TITLE
fix: IOSession access this before calling super

### LINKED ISSUE #40399: IOSession class is accessing this before calling super
**TypeScript Version:**  master branch

I changed TypeScript build target from "ES5" to "ES2019" and it failed to start.

```
 if (this.constructed) {
                    ^

ReferenceError: Must call super constructor in derived class before accessing 'this' or returning from derived constructor
```

image

### PR DESCRIPTION
**Pull request by Jack-Works**
close #40399 

This makes the TS server crashes if the build target is ES6 or higher.

Fixes #40399

### CODE REVIEW

File: src/tsserver/server.ts [hunk 1]
```diff
@@ -484,8 +484,9 @@ namespace ts.server {
         private constructed: boolean | undefined;
```
Comment (sheetalkamat) [inline src/tsserver/server.ts:484]: i think you can get rid of private variable and always use the local variable instead

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: src/tsserver/server.ts (hunk 1 above)
Comment (Jack-Works) [reply src/tsserver/server.ts:484]: this variable is used in other method that cannot access to the local variable

**Comment by sandersn** [comment]
@sheetalkamat Is this ready to go in then?

File: src/tsserver/server.ts [hunk 2]
```diff
@@ -484,8 +484,9 @@ namespace ts.server {
         private constructed: boolean | undefined;
 
         constructor() {
+            let isConstructed = false;
```
Comment (sheetalkamat) [inline src/tsserver/server.ts:487]: Instead of this I would move the logic of scheduling things into method event itself

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**